### PR TITLE
awsappsignals processor: check for case insensitivity in metric names

### PR DESCRIPTION
**Pull request by lisguo**
# Description of the issue
The adot/otel language instrumentation sdks sometimes sends metric names in lowercase over otlp. This PR maps the application signals metrics from lowercase to uppercase (ex: error -> Error)

# Description of changes
Add a map for lowercase metrics to the uppercase ones for emission

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
Passed 
https://github.com/aws/amazon-cloudwatch-agent/actions/runs/7975715500/job/21774963354

# Requirements
_Before commit the code, please do the following steps._
1. Run `make fmt` and `make fmt-sh`
2. Run `make lint`

### CODE REVIEW

File: plugins/processors/awsappsignals/processor.go [hunk 1]
```diff
@@ -132,6 +138,10 @@ func (ap *awsappsignalsprocessor) processMetrics(ctx context.Context, md pmetric
 			metrics := ils.Metrics()
 			for k := 0; k < metrics.Len(); k++ {
 				m := metrics.At(k)
+				// Check if metric name is lowercase
```
Comment (okankoAMZ) [inline plugins/processors/awsappsignals/processor.go:141]: Can't we make all metric names start with a capital letter with `Title()` and not worry about the map?

**Review by okankoAMZ — COMMENTED** [review]
Look! good have a small question

File: plugins/processors/awsappsignals/processor.go [hunk 2]
```diff
@@ -132,6 +136,7 @@ func (ap *awsappsignalsprocessor) processMetrics(ctx context.Context, md pmetric
 			metrics := ils.Metrics()
 			for k := 0; k < metrics.Len(); k++ {
 				m := metrics.At(k)
+				m.SetName(metricCaser.String(m.Name())) // Ensure metric name is uppercase
```
Comment (mitali-salvi) [inline plugins/processors/awsappsignals/processor.go:139]: Nitpick - comment should say its in sentence case, not upper case

**Review by mitali-salvi — DISMISSED** [review]

File: plugins/processors/awsappsignals/processor.go (hunk 2 above)
Comment (okankoAMZ) [reply plugins/processors/awsappsignals/processor.go:139]: +1

**Review by okankoAMZ — DISMISSED** [review]

**Review by mitali-salvi — APPROVED** [review]

**Review by okankoAMZ — APPROVED** [review]

**Review by srprash — APPROVED** [review]

**Pull request merged**